DashWallet/Sources/Infrastructure/Currency Exchanger/Data Provider/RatesProvider.swift

```diff
@@ -61,60 +62,59 @@ protocol RatesProvider: AnyObject {
 // MARK: - RatesProviderFactory
 
 enum RatesProviderFactory {
-    static var base: RatesProvider { BaseRatesProvider() }
+    static var base: RatesProvider = BaseRatesProvider.shared
 }
 
 // MARK: - BaseRatesProvider
 
 final class BaseRatesProvider: NSObject, RatesProvider {
-    private let kRefreshTimeInterval: TimeInterval = 60
-    private let kPriceByCodeKey = "DS_PRICEMANAGER_PRICESBYCODE"
+    private var cancellableBag = Set<AnyCancellable>()
+    static var shared: BaseRatesProvider = BaseRatesProvider()
+    
+    private var lastPriceSourceInfo: String!
+    private let operationQueue: DSOperationQueue
 
     var updateHandler: (([RateObject]) -> Void)? {
         didSet {
-            let plainPricesByCode = UserDefaults.standard.object(forKey: kPriceByCodeKey) as! [String : NSNumber]
-            updateHandler?(plainPricesByCode.map { code, rate in
-                RateObject(code: code, name: currencyName(fromCode: code), price: rate.decimalValue)
-            })
+            self.emitRates()
         }
     }
-
-    private var lastPriceSourceInfo: String!
-
-    private let operationQueue: DSOperationQueue
+    
+    var lastUpdated: Int {
+        get { UserDefaults.standard.integer(forKey: LAST_RATES_RETRIEVAL_TIME) }
+    }
+    
+    @Published private(set) var hasFetchError: Bool = false
+    @Published private(set) var isVolatile: Bool = false
 
     override init() {
         operationQueue = DSOperationQueue()
         super.init()
     }
 
     func startExchangeRateFetching() {
-        updatePrices()
-    }
-
-    @objc
-    func updatePrices() {
-        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(Int(kRefreshTimeInterval))) { [weak self] in
-            self?.updatePrices()
-        }
-
-        let priceOperation = DSPriceOperationProvider.fetchPrices { [weak self] prices, priceSource in
```

Comment:
This code duplicates DashSync and is unnecessary since DashSync is already fetching rates periodically.

While I would prefer exchange rates to be moved from DashSync to the client side, I think duplicating logic isn't the right way to do this - we first need to uncouple rates from DashSync. Right now they are used in forming payment requests.